pkg/tso/keyspace_group_manager.go

```diff
@@ -731,3 +760,69 @@ func genNotServedErr(perr *perrors.Error, keyspaceGroupID uint32) error {
 			"requested keyspace group with id %d %s by this host/pod",
 			keyspaceGroupID, errs.NotServedErr))
 }
+
+// checkTSOSplit checks if the given keyspace group is in split state, and if so, it will make sure the
+// newly split TSO keep consistent with the original one.
+func (kgm *KeyspaceGroupManager) checkTSOSplit(
+	keyspaceGroupID uint32,
+	dcLocation string,
+) error {
+	splitAM, splitGroup := kgm.getKeyspaceGroupMeta(keyspaceGroupID)
```

Comment:
What if there are multiple TSO requests coming in and they all pass line 804 and cause splitAllocator.SetTSO() and kgm.finishSplitKeyspaceGroup to be invoked multiple times? Do we need to put the function into a critical section?

Reply:
It's not a problem since `SetTSO` can only set the TSO to a bigger value.

Reply:
That part is fine. My real worry is that if tso request concurrency is high, it will cause finishSplitKeyspaceGroup to be invoked too many times, which isn't cheap and result in transaction conflicts and high load on etcd, though one transaction will succeed eventually.

Reply:
I'll sign off for now. Suggest to make some protection here.

Reply:
OK, I will make some refinements here today.